### ISSUE TITLE
Need the ability to prevent event handlers from be called before a particular time.

### ISSUE THREAD

**Issue by dchassin** [gridlab-d/gridlab-d#1133]
Some users need to prevent events from being called until a particular update time, e.g., external implementations of market clearing mechanisms.  It's usually easy to block events within the handlers, but when scripts are being called at the object level (see #1132) this becomes a very serious performance problem.  

An attempt to implement a solution using the _validto_ object header property required the addition of a global variable called _validto\_context_ to establish in what context events get suppressed by the value of _validto_.  This seems like a weak solution, and it may not be correct at all given the purpose of _validto_.  See https://github.com/dchassin/gridlabd/commit/33d3653fd07d1d615a7e86cf03e163c00e28ca5a for details.

More likely the addition of a new header property to suppress event calls until a particular time may be required.  It's not at all clear if this would interfere destructively with the general behavior of event handlers.

Recommend further discussion on how to implement such a feature.

**Comment by dchassin** [comment]
This is required for the Powernet CEC project

**Comment by mlamlamla** [comment]
Indeed, validto_context does not fulfill the requirements as needed.

We would need a functionality which allows different call frequencies for independent objects. Example: We have a market and a powerflow layer in GridlabD. The market (and the associated external script called by on_precommit) needs to run every 5min, the powerflow every 0.5min.

Thanks for your suggestions how to handle that efficiently!

**Comment by dchassin** [comment]
I will work further experiments with you on https://github.com/dchassin/gridlabd and report back when we have something that is safe and effective.
